lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -67,59 +67,27 @@ ResultCode HostManager::addNetwork(Module *M) {
   }
   auto partitioner = Partitioner(M, deviceInfo);
   auto nodeList = std::move(partitioner.Partition());
-  auto result = provisioner_->provision(nodeList.roots, *M);
+  auto result = provisioner_->provision(nodeList, *M);
 
-  for (auto &node : nodeList.roots) {
-    roots_.emplace(node->name, std::move(node));
-  }
-  for (auto &node : nodeList.nodes) {
-    networks_.emplace(node->name, std::move(node));
+  for (auto &node : nodeList) {
+    networks_.emplace((node.first)->name, std::move(node));
   }
   return result;
 }
 
 void HostManager::removeNetwork(llvm::StringRef networkName) {
-  // Walk the tree for the given function, calling evict function for each node
-  // before removing it from networks_ which frees the node.
   std::lock_guard<std::mutex> networkLock(networkLock_);
-  if (roots_.find(networkName) == roots_.end()) {
+  auto networkIterator = networks_.find(networkName);
+  if (networkIterator == networks_.end()) {
     return;
   }
-  std::queue<std::string> nodes;
-  std::set<std::string> allNodes;
-  for (auto &child : roots_[networkName]->children) {
-    nodes.push(child->name);
-    allNodes.insert(child->name);
-  }
-
-  while (!nodes.empty()) {
-    auto nodeName = nodes.front();
-    nodes.pop();
-    auto it = networks_.find(nodeName);
-    if (it != networks_.end()) {
-      for (auto &node : it->second->children) {
-        auto name = node->name;
-        if (allNodes.find(node->name) == allNodes.end()) {
-          nodes.push(node->name);
-          allNodes.insert(node->name);
-        }
-      }
-    }
-  }
-  roots_.erase(roots_.find(networkName));
-  for (auto &networkName : allNodes) {
-    auto networkIterator = networks_.find(networkName);
-    if (networkIterator != networks_.end()) {
-      devices_[networkIterator->second->deviceID]->evictNetwork(
-          networkName, /*evictCB=*/nullptr);
-      networks_.erase(networkIterator);
-    }
-  }
+  networks_[networkName].second.clear();
```

Comment:
Do we need to clear the vector first? If we just call erase shouldn't that be enough?